Modules/Brand/Http/Services/BrandService.php

```diff
@@ -206,15 +228,63 @@ public function getShop(string $brandKey): array
         $brand->country = $country->name;
         //headquarter
-        $headquarteredCountry = Country::where('id', $brand->headquatered)->first();
+        $headquarteredCountry = Country::where('id', $brand->headquatered)->first(['name']);
         $brand->headquatered = $headquarteredCountry->name;
         //shipped from
-        $productShippedCountry = Country::where('id', $brand->product_shipped)->first();
+        $productShippedCountry = Country::where('id', $brand->product_shipped)->first(['name']);
         $brand->product_shipped = $productShippedCountry->name;
 
+        //shop-wide promotion banner
+        $promotions = Promotion::where('user_id', $brand->user_id)
+            ->where('promotion_type', 'order')
+            ->where('status', 'active')
+            ->where('from_date', '<=', date('Y-m-d'))
+            ->where('to_date', '>=', date('Y-m-d'))
+            ->orderBy('discount_amount', 'ASC')
+            ->get(['discount_amount','discount_type','ordered_amount']);
+        $shopPromotions = [];
+        $remaining_days = 0;
+        if (!empty($promotions)) {
+            if (count($promotions) > 1) {
+                $promotionDetails = Promotion::where('user_id', $brand->user_id)
+                    ->where('promotion_type', 'order')
+                    ->where('status', 'active')
+                    ->where('from_date', '<=', date('Y-m-d'))
+                    ->where('to_date', '>=', date('Y-m-d'))
+                    ->orderBy('discount_amount', 'DESC')
+                    ->first(['discount_amount','discount_type']);
+                if ($promotionDetails->discount_type === 1) {
+                    $maxDiscountAmount = $promotionDetails->discount_amount . '%';
+                } else {
+                    $maxDiscountAmount = '$' . $promotionDetails->discount_amount;
+                }
+                $shopPromotions[] = 'Up to ' . $maxDiscountAmount . ' off';
+            }
+            foreach ($promotions as $promotion) {
+                if ($promotion->discount_type === 1) {
+                    $discountAmount = $promotion->discount_amount . '%';
+                } else {
+                    $discountAmount = '$' . $promotion->discount_amount;
+                }
+                $shopPromotions[] = $discountAmount . ' off orders $' . $promotion->ordered_amount . '+';
+            }
+            $promotionDetails = Promotion::where('user_id', $brand->user_id)
+                ->where('promotion_type', 'order')
+                ->where('status', 'active')
+                ->where('from_date', '<=', date('Y-m-d'))
+                ->where('to_date', '>=', date('Y-m-d'))
+                ->orderBy('to_date', 'DESC')
+                ->first(['to_date']);
+            $date = $promotionDetails->to_date;
+            $remaining_days = now()->diffInDays(Carbon::parse($date));
+        }
+        $brand->promotions = $shopPromotions;
+        $brand->promotion_remaining_days = $remaining_days;
```

Comment:
Use this code instead of that:
$promotions = Promotion::where('user_id', $brand->user_id)
    ->where('promotion_type', 'order')
    ->where('status', 'active')
    ->whereDate('from_date', '<=', date('Y-m-d'))
    ->whereDate('to_date', '>=', date('Y-m-d'))
    ->orderBy('discount_amount', 'ASC')
    ->get(['discount_amount', 'discount_type', 'ordered_amount']);

$shopPromotions = [];
$remaining_days = 0;

if ($promotions->isNotEmpty()) {
    $promotionDetails = $promotions->sortByDesc('discount_amount')->first(['discount_amount', 'discount_type']);

    if ($promotionDetails->discount_type === 1) {
        $maxDiscountAmount = $promotionDetails->discount_amount . '%';
    } else {
        $maxDiscountAmount = '$' . $promotionDetails->discount_amount;
    }

    if ($promotions->count() > 1) {
        $shopPromotions[] = 'Up to ' . $maxDiscountAmount . ' off';
    }

    foreach ($promotions as $promotion) {
        if ($promotion->discount_type === 1) {
            $discountAmount = $promotion->discount_amount . '%';
        } else {
            $discountAmount = '$' . $promotion->discount_amount;
        }

        $shopPromotions[] = $discountAmount . ' off orders $' . $promotion->ordered_amount . '+';
    }

    $promotionDetails = $promotions->sortByDesc('to_date')->first(['to_date']);
    $remaining_days = Carbon::parse($promotionDetails->to_date)->diffInDays(now());
}

$brand->promotions = $shopPromotions;
$brand->promotion_remaining_days = $remaining_days;
**Make sure to test it before pushing.**